app/state/reducers/dataReducer.js

```diff
@@ -0,0 +1,67 @@
+import capitalize from 'lodash/capitalize';
+import keyBy from 'lodash/keyBy';
+
+import actionTypes from 'actions/actionTypes';
+
+const initialState = {
+  units: {},
+};
+
+function normalizeOwner(owner) {
+  switch (owner.toLowerCase()) {
+    case 'avain asumisoikeus':
+    case 'avain asumisoikeus oy':
+      return 'AVAIN Asumisoikeus Oy';
+    case 'asuntosäätiön asumisoikeus oy':
+      return 'Asuntosäätiön Asumisoikeus Oy';
+    case 'helsingin seudun asumisoikeusyhdistys':
+    case 'helsingin seudun asumisoikeusyhdistys helas':
+      return 'Helsingin Seudun Asumisoikeusyhdistys HELAS';
+    case 'ta-asumisoikeus':
+    case 'ta-asumisoikeus oy':
+      return 'TA-Asumisoikeus Oy';
+    default:
+      return owner;
+  }
+}
+
+function getUnitOwnerAndName(data) {
+  const dataName = data.name.fi;
+  const slashIndex = dataName.indexOf('/');
+  if (slashIndex !== -1) {
+    const owner = normalizeOwner(dataName.substring(0, slashIndex));
+    const name = dataName.substring(slashIndex + 1);
+    return { owner, name };
+  }
+  return {
+    owner: normalizeOwner(dataName),
+    name: data.street_address.fi,
+  };
+}
+
+function getUnit(data) {
```

Comment:
- [ ] There could be test cases for the different data fields. For example that `streetAddress` is gotten from data.street_address.fi and that `city` is gotten from `data.municipality` and capitalized etc.